kvdb/postgres/db.go

```diff
@@ -0,0 +1,161 @@
+package postgres
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/btcsuite/btcwallet/walletdb"
+	"github.com/jackc/pgx/v4/pgxpool"
+)
+
+// KV stores a key/value pair.
+type KV struct {
+	key string
+	val string
+}
+
+// db holds a reference to the postgres connection connection.
+type db struct {
+	cfg  Config
+	ctx  context.Context
+	pool *pgxpool.Pool
+}
+
+// Enforce db implements the walletdb.DB interface.
+var _ walletdb.DB = (*db)(nil)
+
+// newPostgresBackend returns a db object initialized with the passed backend
+// config. If etcd connection cannot be estabished, then returns error.
+func newPostgresBackend(ctx context.Context, cfg Config) (*db, error) {
+	pool, err := pgxpool.Connect(ctx, cfg.Dsn)
+	if err != nil {
+		return nil, fmt.Errorf("unable to connect to database: %v", err)
+	}
+
+	_, err = pool.Exec(context.TODO(), `
```

Comment:
Yes we could. I purposefully left all contexts to TODO() because we need to decide what we want to do with this. The kvdb abstraction doesn't allow for a context to be passed anywhere, so we'll probably have to create our own contexts with a timeout. This also ties into the question what to do when a context times out for kvdb calls that don't have an error return value (`Get()`).

Reply:
My suggestion after the latest offline discussion:

We have the main context passed in through the driver and extend it with a configurable timeout. We probably have to define these as pretty long default timeouts and use different ones for read and write.
If we run into an expired context error, the `panic() ... recover()` pattern that we discussed should then handle it.

Reply:
Implemented `panic..recover` and discovered quite a few more cases where errors were suppressed. Left it as a separate commit for now for easy review.